### PR TITLE
Ankit | Stage | Fetch Go cardless integrated bank balance in dashboard page

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cx4gg7h

### CODE REVIEW

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.ts [hunk 1]
```diff
@@ -69,6 +82,27 @@ export class BankAccountsComponent implements OnInit, OnDestroy {
                 this.getAccounts(this.fromDate, this.toDate, 'bankaccounts', null, null, 'true', 20, '', 'closingBalance', 'desc');
             }
         };
+
+        this.store.pipe(select(profileObj => profileObj.settings.profile), takeUntil(this.destroyed$)).subscribe((profile) => {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.ts]: Write this in component store

File: apps/web-giddh/src/app/home/home.store.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    *  @memberof HomeComponentStore
+    */
+    readonly refreshBank = this.effect((data: Observable<void>) => {
+        return data.pipe(
+            switchMap(() => {
+                this.patchState({ bankMessage: null, isBankRefreshing: true});
+                return this.contactService.refreshBank().pipe(
+                    tapResponse(
+                        (res: any) => {
+                            if (res?.status === "success" && res?.body) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/home.store.ts:42]:
```suggestion
                            if (res?.status === "success" && res.body) {
```

Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/home.store.ts:42]:
```suggestion
                            if (res.status === "success" && res?.body) {
```

File: apps/web-giddh/src/app/home/home.store.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                                this.toaster.showSnackBar("success", res.body);
+                                return this.patchState({ bankMessage: res.body, isBankRefreshing: false});
+                            } else {
+                                res?.message && this.toaster.showSnackBar("error", res.message);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/home.store.ts:46]:
```suggestion
                                res.message && this.toaster.showSnackBar("error", res.message);
```

File: apps/web-giddh/src/app/services/contact.service.ts [hunk 4]
```diff
@@ -121,4 +126,21 @@ export class ContactService {
             }), catchError((e) => this.errorHandler.HandleCatch<any, string>(e, '', '')));
         }
     }
+
+     /**
+     * Refresh bank accounts
+     * 
+     * @returns {Observable<BaseResponse<any, any>>}
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/contact.service.ts]: check doc and fix Observable type

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/contact.service.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/services/contact.service.ts]: Done

**Review by dvCodeWorld — DISMISSED** [review]

File: apps/web-giddh/src/app/home/home.store.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import { Store } from "@ngrx/store";
+
+export interface HomeState {
+    bankMessage: any;
+    isBankRefreshing: boolean;
+}
+
+const DEFAULT_STATE: HomeState = {
+    bankMessage: null,
+    isBankRefreshing: false,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/home/home.store.ts]:
```suggestion
    isBankRefreshing: false
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

**Review by dilpreetsinghofficial — DISMISSED** [review]

File: apps/web-giddh/src/app/services/contact.service.ts [hunk 6]
```diff
@@ -121,4 +126,21 @@ export class ContactService {
...
         }
     }
+
+     /**
+     * Refresh bank accounts
+     * 
+     * @returns {Observable<BaseResponse<IBankRefreshResponse, any>>}
+     * @memberof ContactService
+     */
+     public refreshBank(): Observable<BaseResponse<IBankRefreshResponse, any>> {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/contact.service.ts]:
```suggestion
     public refreshGoCardlessBankTransactions(): Observable<BaseResponse<IBankRefreshResponse, any>> {
```

File: apps/web-giddh/src/app/services/apiurls/contact.api.ts [hunk 7]
```diff
@@ -1,3 +1,4 @@
 export const CONTACT_API = {
-    ADD_COMMENT: 'v2/company/:companyUniqueName/accounts/:accountUniqueName/report-comment'
+    ADD_COMMENT: 'v2/company/:companyUniqueName/accounts/:accountUniqueName/report-comment',
+    BANKACCOUNTS_REFRESH: 'company/:companyUniqueName/gocardless/refresh'
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/apiurls/contact.api.ts]:
```suggestion
    GOCARDLESS_BANKTRANSACTIONS_REFRESH: 'company/:companyUniqueName/gocardless/refresh'
```

File: apps/web-giddh/src/app/home/home.store.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    ) {
+        super(DEFAULT_STATE);
+    }
+    public profile$: Observable<any> = this.select(this.store.select(profileObj => profileObj.settings.profile), (response) => response);
+    /**
+    *  Refresh bank accounts
+    *
+    *  @memberof HomeComponentStore
+    */
+    readonly refreshBank = this.effect((data: Observable<void>) => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/home/home.store.ts]:
```suggestion
    readonly refreshGoCardlessBankTransactions = this.effect((data: Observable<void>) => {
```

File: apps/web-giddh/src/app/services/contact.service.ts [hunk 9]
```diff
@@ -121,4 +126,21 @@ export class ContactService {
             }), catchError((e) => this.errorHandler.HandleCatch<any, string>(e, '', '')));
         }
     }
+
+     /**
+     * Refresh bank accounts
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/contact.service.ts]: Update js doc also

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/contact.service.ts (hunk 6 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/services/contact.service.ts]: Done

File: apps/web-giddh/src/app/home/home.store.ts (hunk 8 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/home/home.store.ts]: Done

File: apps/web-giddh/src/app/services/apiurls/contact.api.ts (hunk 7 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/services/apiurls/contact.api.ts]: Done

File: apps/web-giddh/src/app/services/contact.service.ts (hunk 9 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/services/contact.service.ts]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**